forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -32,84 +30,60 @@ internal object LDContextKind {
-        }
+    internal fun initialize(app: Application, ldConfig: LDConfig) {
         val contextKind = ContextKind.of(LDContextKind.SERVICE)
         val context = LDContext.create(contextKind, LDContexts.ANDROID_CONTEXT)
         client = LDClient.init(app, ldConfig, context, 0)
-
-        internalLogger = logger
     }
 
-    // vaultType is instantiated lazily and is a singleton. Once we set the vault type once, we don't
-    // want to overwrite it! We must take in the application as a parameter, which means that a
-    // ForagePINEditText must be rendered before any of the ForageSDKApi functions are called.
-    internal fun getVaultProvider(): VaultType {
-        if (vaultType != null) {
-            return vaultType as VaultType
-        }
-        // default to 100% VGS usage in case LD flag retrieval fails
-        val defaultVal = 0.0
-        val vaultPercent =
-            client?.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, defaultVal) ?: defaultVal
-        internalLogger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
-        val randomNum = Math.random() * 100
-
-        vaultType = if (randomNum < vaultPercent) {
-            VaultType.BT_VAULT_TYPE
-        } else {
-            VaultType.VGS_VAULT_TYPE
-        }
-        internalLogger.i("[LaunchDarkly] Vault type set to $vaultType")
-        return vaultType as VaultType
+    internal fun getVaultProvider(logger: Log = Log.getSilentInstance()): VaultType {
+        val vaultPercent = client?.doubleVariation(
+            LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG,
+            ALWAYS_VGS_PERCENT
+        ) ?: ALWAYS_VGS_PERCENT
+        logger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
+
+        // convert the flag percent into an answer to which vault provider to use
+        val vaultType = computeVaultType(vaultPercent)
+        logger.i("[LaunchDarkly] Vault type set to $vaultType")
+
+        // return vault provider derived from the novel or cached flag value
+        return vaultType
     }
 
-    internal fun getPollingIntervals(): LongArray {
+    internal fun getPollingIntervals(logger: Log = Log.getSilentInstance()): LongArray {
         val defaultVal = LDValue.buildObject().put(
             "intervals",
             LDValue.Convert.Long.arrayFrom(List(10) { 1000L })
         ).build()
 
-        val jsonIntervals = client?.jsonValueVariation(LDFlags.ISO_POLLING_WAIT_INTERVALS, defaultVal) ?: defaultVal
+        val jsonIntervals = client?.jsonValueVariation(
+            LDFlags.ISO_POLLING_WAIT_INTERVALS,
+            defaultVal
+        ) ?: defaultVal
 
         val intervals = jsonIntervals.get("intervals")
 
         // Converting the LDArray into a LongArray
         val pollingList = LongArray(intervals.size()) { intervals.get(it).longValue() }
 
-        internalLogger.i("[LaunchDarkly] polling intervals $pollingList")
+        logger.i("[LaunchDarkly] polling intervals $pollingList")
```

Comment:
[This](https://app.datadoghq.com/logs?query=service%3Aandroid-sdk%20%22%5BLaunchDarkly%5D%20polling%20intervals%22%20&cols=host%2Cservice&index=%2A&messageDisplay=inline&refresh_mode=paused&stream_sort=desc&viz=stream&from_ts=1699602086056&to_ts=1699623600000&live=false) is what the log of polling intervals currently looks like :( looks like we need some kind of `toString` method?

Reply:
roger that! If it's cool with you, I made a follow on ticket for this [here](https://linear.app/joinforage/issue/FX-718/create-a-tostring-method-for-dollarpollinglist) just to keep that fix isolated